src/main/scala/com/kinja/amqp/MessageBufferProcessor.scala

```diff
@@ -92,10 +92,11 @@ class MessageBufferProcessor(
 		val batchSizeWithExtraGap = batchSize * 2
 		val messages = messageStore.loadLockedMessages(batchSizeWithExtraGap)
 		scala.util.Random.shuffle(producers).foreach {
-			case (exchange, producer) =>
-				val messagesToProducer = messages.filter(_.exchangeName == producer.exchange.name)
+			case (exchange: String, producer: AmqpProducerInterface) =>
```

Comment:
It does not, try to change it to a concrete implementation and compile

On Monday, 14 March 2016, Zoltán Balázs <notifications@github.com> wrote:

> In src/main/scala/com/kinja/amqp/MessageBufferProcessor.scala
> <https://github.com/gawkermedia/amqp-client-provider/pull/23#discussion_r55991329>
> :
>
> > @@ -92,10 +92,11 @@ class MessageBufferProcessor(
> >  		val batchSizeWithExtraGap = batchSize * 2
> >  		val messages = messageStore.loadLockedMessages(batchSizeWithExtraGap)
> >  		scala.util.Random.shuffle(producers).foreach {
> > -			case (exchange, producer) =>
> > -				val messagesToProducer = messages.filter(_.exchangeName == producer.exchange.name)
> > +			case (exchange: String, producer: AmqpProducerInterface) =>
>
> I think this change results in isInstanceOfcalls which is not necessary.
> You could keep this line unchanged.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gawkermedia/amqp-client-provider/pull/23/files#r55991329>
> .
>